src/pathwayvis/services/actions/actions.service.ts

```diff
@@ -125,6 +125,36 @@ class UndoSetObjective extends SetObjective {
   }
 }
 
+@registerAction
+// tslint:disable-next-line
+class ChangeBounds extends ReactionAction implements Action {
+  public label = 'Change bounds';
+  public type: string = 'reaction:bounds:do';
+  public shared: types.MapData;
+
+  public canDisplay(context) {
+    const isRemoved = context.shared.changedReactions.find((reaction) => reaction.id === context.element.bigg_id);
+    return context.type === 'map:reaction' && !isRemoved;
+  }
+}
+
+@registerAction
+// tslint:disable-next-line
+class UndoChangeBounds extends ChangeBounds {
+  public label = 'Undo change bounds';
+  public type: string = 'reaction:bounds:undo';
+
+  public canDisplay(context) {
+    if (context.shared.changedReactions) {
```

Comment:
So can this be `null`? I'd prefer to leave missing arrays as empty lists.

Reply:
It's only undefined if find cant found any object.

Reply:
Where do you set it?